### PR TITLE
[in-development] Bug 774497 — User messaging for Fennec

### PR DESCRIPTION
**Pull request by rnewman**
Development work ongoing for https://bugzilla.mozilla.org/show_bug.cgi?id=774497.

### CODE REVIEW

**Comment by ncalexan** [comment]
r=nalexander for Part 0 and Part 1.

File: manifests/SyncAndroidManifest_activities.xml.in [hunk 1]
```diff
@@ -75,3 +75,25 @@
                 <data android:mimeType="text/*" />
             </intent-filter>
         </activity>
+
```
Comment (ncalexan) [inline manifests/SyncAndroidManifest_activities.xml.in]: Separate manifest files, plz!  Announcements have nothing to do with Sync.

File: src/main/java/org/mozilla/gecko/sync/GlobalConstants.java.in [hunk 2]
```diff
@@ -10,10 +10,29 @@ package org.mozilla.gecko.sync;
  */
 public class GlobalConstants {
   public static final String PRODUCT_NAME = "@MOZ_APP_DISPLAYNAME@";
-  public static final String SYNC_VERSION_STRING = "1.@MOZ_APP_VERSION@.0";
+  public static final String ANDROID_CPU_ARCH = "@ANDROID_CPU_ARCH@";
```
Comment (ncalexan) [inline src/main/java/org/mozilla/gecko/sync/GlobalConstants.java.in:13]: Time to separate out some of this stuff into a non-Sync specific `org/mozilla/gecko/Constants.java.in` or similar?  I really think we should take this time to start re-organizing some of the Sync code for generalization to new services.

File: src/main/java/org/mozilla/gecko/background/BackgroundServiceConstants.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  static final String PREFS_BRANCH = "background";
+  static final String PREF_LAST_FETCH  = "last_fetch";
+  static final String PREF_LAST_LAUNCH = "last_firefox_launch";
+  static final String PREF_ANNOUNCE_SERVER_URL  = "announce_server_url";
+  static final String PREF_EARLIEST_NEXT_ANNOUNCE_FETCH = "earliest_next_announce_fetch";
+  static final String PREF_ANNOUNCE_FETCH_INTERVAL_MSEC = "announce_fetch_interval_msec";
+
+  public static final int SHARED_PREFERENCES_MODE = 0;
+
+  static final String DEFAULT_ANNOUNCE_SERVER_URL = "http://people.mozilla.com/~rnewman/announce/";
```
Comment (ncalexan) [inline src/main/java/org/mozilla/gecko/background/BackgroundServiceConstants.java:24]: Urgh.

**Comment by ncalexan** [comment]
General comments:
- There are a few Javadoc warnings.  Please fix before landing.
- We expect multiple background modules, right?  So let's forward engineer a little and put things in `background.announcements`, and separate the `BackgroundServiceConstants` from the `AnnouncementsConstants`.  One of my biggest complaints about Sync is that the code is haphazardly strewn throughout the namespace.  You could consider separating the `.net` code from the `.services`, as well, but with a relatively tightly scoped feature like this, I'm not going to push it :)

This is looking great, really solid engineering.  I love reviewing clean new code!

**Comment by ncalexan** [comment]
I think this is deprecated in favour of a later pull request.  rnewman: please close if I am correct.

**Comment by rnewman** [comment]
Yup, and I think I've addressed all of the comments, too.

**Pull request closed without merging**